pkg/build/build.go

```diff
@@ -143,7 +143,15 @@ func CreateTarball(source, target string) error {
 			if err != nil {
 				return err
 			}
-			header, err := tar.FileInfoHeader(info, info.Name())
+			var link string
+			if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+
+				if link, err = os.Readlink(path); err != nil {
```

Comment:
nested if statement, may be better with `if info.Mode()&os.ModeSymlink == os.ModeSymlink &&  link, err = os.Readlink(path); err != nil`

Reply:
didn't work :(

Reply:
@cdrage I got following error, 
```
pkg/build/build.go:148:52: invalid operation: info.Mode() & os.ModeSymlink == os.ModeSymlink && symLink (mismatched types bool and string)
```

Reply:
Post the full code.  This doesn't look like the full-line if-statement messed it up, but rather the `info.Mode()&os.ModeSymlink == os.ModeSymlink` part.